lib/SILOptimizer/Mandatory/DiagnoseInfiniteRecursion.cpp

```diff
@@ -94,37 +94,39 @@ static bool hasRecursiveCallInPath(SILBasicBlock &Block,
   return false;
 }
 
-static bool hasInfinitelyRecursiveApply(SILFunction &Fn,
-                                        SILFunction *TargetFn) {
+/// Perform a DFS through the target function to find any paths to an exit node
+/// that do not call into the target.
+static bool hasInfinitelyRecursiveApply(SILFunction *TargetFn) {
   SmallPtrSet<SILBasicBlock *, 16> Visited;
-  SmallVector<SILBasicBlock *, 16> WorkList;
-  // Keep track of whether we found at least one recursive path.
-  bool foundRecursion = false;
+  SmallVector<SILBasicBlock *, 16> WorkList = { TargetFn->getEntryBlock() };
```

Comment:
Why not do 32 instead of 16? Seems better for medium sized functions and you aren't going to have multiple invocations of this function so the stack usage isn't too crazy to me.